### PR TITLE
feat: support DesignToken in SSR

### LINKED ISSUE #6287: feat: add DesignToken support to SSR DOM shim
# 🙋 Feature Request
To support DesignToken in SSR, several APIs need to be fleshed-out in the exposed DOM shim:
1. CSSStyleSheet.insertRule()
2. CSSRuleList
3. CSSStyleRule
4. CSSStyleDeclaration (the .setProperty() and .removeProperty() methods)

### PR DESCRIPTION
**Pull request by nicholasrice**
# Pull Request

## 📖 Description
This PR adds support for DesignToken in SSR environments. It does this by implementing a resolution strategy that uses custom events (supported in SSR) instead of manual DOM walking (not supported in SSR) to construct DesignTokenNode hierarchies. It also fleshes out the SSR DOM shim's CSSOM to support setting and removing properties on CSSStyleRules, which are used by DesignToken internally. 

### 🎫 Issues
closes #6287 #6272

### General

- [ ] I have included a change request file using `$ yarn change`
- [ ] I have added tests for my changes.
- [ ] I have tested my changes.
- [ ] I have updated the project documentation to reflect my changes.
- [ ] I have read the [CONTRIBUTING](https://github.com/microsoft/fast/blob/master/CONTRIBUTING.md) documentation and followed the [standards](/docs/community/code-of-conduct/#our-standards) for this project.

### Component-specific

- [ ] I have added a new component
- [ ] I have modified an existing component
- [ ] I have updated the [definition file](https://github.com/microsoft/fast/blob/master/packages/web-components/fast-components/CONTRIBUTING.md#definition)
- [ ] I have updated the [configuration file](https://github.com/microsoft/fast/blob/master/packages/web-components/fast-components/CONTRIBUTING.md#configuration)

### CODE REVIEW

File: packages/web-components/fast-ssr/src/dom-shim.ts [hunk 1]
```diff
@@ -119,14 +119,48 @@ class CustomEvent<T = any> extends Event {
...
 /**
  * @beta
  */
 export class CSSStyleSheet {
     replace() {}
     public readonly cssRules: CSSRule[] = [];
-    insertRule(rule: CSSRule, index: number = 0) {
-        this.cssRules.splice(index, 0, rule);
+    insertRule(rule: string, index: number = 0) {
+        const selectorText = rule.split("{")[0];
```
Comment (radium-v) [inline packages/web-components/fast-ssr/src/dom-shim.ts:158]: Is this safe for media queries?

Comment (nicholasrice) [reply packages/web-components/fast-ssr/src/dom-shim.ts:158]: No. This shim isn't supposed to be comprehensive, and there are many APIs that are incomplete or don't exist on these objects. Is there a core case we need to handle that includes writing media queries w/ insertRule?

**Review by EisenbergEffect — APPROVED** [review]
Is there a place where this gets plugged in? Do we need to update the Todo SSR example to show this?

File: packages/web-components/fast-ssr/src/dom-shim.ts (hunk 1 above)
Comment (radium-v) [reply packages/web-components/fast-ssr/src/dom-shim.ts:158]: > Is there a core case we need to handle that includes writing media queries w/ insertRule?

Not that I'm familiar with - I just like to double-check when I see string manipulation used to assist code evaluation.

**Comment by nicholasrice** [comment]
@EisenbergEffect Yeah I think that is a good idea. There is a task here: #6292. I also plan to address SSR in https://github.com/microsoft/fast/issues/6273

File: packages/web-components/fast-ssr/src/dom-shim.ts (hunk 1 above)
Comment (nicholasrice) [reply packages/web-components/fast-ssr/src/dom-shim.ts:158]: Gotcha, sounds good. The shim is extensible for anyone that needs a more robust implementation of these APIs (or they can choose to use a different shim all together) - the goal of *this* shim is to cover the API usage of fast-element and fast-foundation.

**Review by radium-v — APPROVED** [review]

**Pull request merged**